### ISSUE TITLE
Improve doc about struct Resolver in var PreferGo

### ISSUE THREAD

**Issue by renanbastos93** [golang/go#35561]
Guys, how are you?

I have a suggestion to improve doc about parameter `PreferGo` on struct Resolver based on a comment in another issue. https://github.com/golang/go/issues/19268#issuecomment-345384876

Suggestion adds more a small text:
```
When you use the variable PreferGo set to true you will be 
able to use your dial, If you use PreferGo set to false it will ignore 
your dial and use directly "/etc/resolv.conf "
```

Reference Doc:
https://github.com/golang/go/blob/a38a917aee626a9b9d5ce2b93964f586bf759ea0/src/net/lookup.go#L120

**Comment by ianlancetaylor** [comment]
If I understand you correctly, you are suggesting that we clearly document that if `PreferGo` is set to `false`, then the `Dial` method in the `Resolver` will not be used.  Is that correct?

**Comment by renanbastos93** [comment]
> If I understand you correctly, you are suggesting that we clearly document that if `PreferGo` is set to `false`, then the `Dial` method in the `Resolver` will not be used. Is that correct?

Hello, @ianlancetaylor 

I am so sorry for my English. I'm in a constant improvement in English skills.
Well, yes correctly. When set false in `PreferGo` it will ignore your Dial and will be using a default.

**Comment by renanbastos93** [comment]
I was thinking if you agreed to this improvement, I can open a pull request.

**Comment by ianlancetaylor** [comment]
I agree with the improvement in general, but I don't agree with the text proposed in the original message.

**Comment by renanbastos93** [comment]
> I agree with the improvement in general, but I don't agree with the text proposed in the original message.

Ok, no problem.
What's your suggestion?

**Comment by ianlancetaylor** [comment]
I'm sorry, I don't have one right now.  It will take some thought and this doesn't seem urgent.

**Comment by ozn0417** [comment]
@ianlancetaylor , I see the documentation is updated (at https://golang.org/pkg/net/#Resolver) indicting that "PreferGo" is a switch between Go's built-in DNS resolver or CGO resolver (which uses "getaddrinfo" or "getnameinfo"). Does using CGO resolver disable us using DNS dialer? Suppose I want to use CGO resolver with custom DNS server.

**Comment by ianlancetaylor** [comment]
@ozn0417 Yes, using the cgo resolver means that it will use the DNS servers listed in `/etc/resolv.conf` or equivalent, and will not use whatever nameservers you want to specify.  This should ideally also be part of the documentation.
